pkg/v1/layout/write.go

```diff
@@ -124,10 +124,14 @@ func (l Path) AppendDescriptor(desc v1.Descriptor) error {
 		return err
 	}
 
-	return l.writeFile("index.json", rawIndex)
+	return l.WriteFile("index.json", rawIndex)
 }
 
-func (l Path) writeFile(name string, data []byte) error {
+// WriteFile write a file with arbitrary data at an arbitrary location in a v1
+// layout. Used mostly internally to write files like "oci-layout" and
+// "index.json", also can be used to write other arbitrary files. Do *not* use
+// this to write blobs. Use only WriteBlob() for that.
+func (l Path) WriteFile(name string, data []byte) error {
```

Comment:
Let's expose the `perm os.FileMode` parameter here to match `ioutil.WriteFile`, in case we don't always want `os.ModePerm`.

Otherwise LGTM.

Reply:
That's a good idea. Doing it now.

Reply:
Done. I had to then add the `os.ModePerm` to all of the other calls to it, but that was easy.